st2auth/st2auth/controllers/v1/auth.py

```diff
@@ -47,6 +49,25 @@ def post(self, request, **kwargs):
         elif cfg.CONF.auth.mode == 'standalone':
             return self._handle_standalone_auth(request=request, **kwargs)
 
+    @jsexpose(arg_types=[str])
+    def get_one(self, token):
+        try:
+            token_db = get_token(token)
+
+            # Check if token has expired.
+            if token_db.expiry < date_utils.get_datetime_utc_now():
```

Comment:
IIRC, there is already `validate_auth_token` method in `st2common/st2common/util/auth.py` so we should ideally re-use that if possible (and if necessary, make changes so we can re-use it).

I think that's especially important since it's security sensitive code.